sound/soc/sof/ipc4-topology.c

```diff
@@ -1668,36 +1762,47 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 
 	available_fmt->ref_audio_fmt = &available_fmt->base_config->audio_fmt;
 
-	/*
-	 * Output format is optional for process modules.
-	 * Process modules setup the output format based on audio format tokens in topology.
-	 */
-	if (process->init_payload_format == INIT_PAYLOAD_WITH_OUTPUT_FMT)
-		ret = sof_ipc4_init_audio_fmt(sdev, swidget, &process->base_config,
-					      &process->output_format, pipeline_params,
-					      available_fmt,
-					      sizeof(struct sof_ipc4_base_module_cfg));
-	else
-		ret = sof_ipc4_init_audio_fmt(sdev, swidget, &process->base_config,
-					      NULL, pipeline_params, available_fmt,
-					      sizeof(struct sof_ipc4_base_module_cfg));
+	ret = sof_ipc4_init_audio_fmt(sdev, swidget, &process->base_config,
+				      &process->output_format, pipeline_params,
+				      available_fmt, sizeof(struct sof_ipc4_base_module_cfg));
 	if (ret < 0)
 		return ret;
 
 	/* update pipeline memory usage */
 	sof_ipc4_update_pipeline_mem_usage(sdev, swidget, &process->base_config);
 
 	/*
-	 * ipc_config_data is composed of the base_config, optional output formats followed
-	 * by the data required for module init in that order.
+	 * ipc_config_data is composed of the base_config, optional output format or optional
+	 * base config extension followed by module specific data in that order
 	 */
 	memcpy(cfg, &process->base_config, sizeof(struct sof_ipc4_base_module_cfg));
 	cfg += sizeof(struct sof_ipc4_base_module_cfg);
 
-	/* copy output format to configure data payload */
-	if (process->init_payload_format == INIT_PAYLOAD_WITH_OUTPUT_FMT) {
+	switch (process->init_payload_format) {
+	case INIT_PAYLOAD_WITH_OUTPUT_FMT:
+		/* copy output format to configure data payload */
 		memcpy(cfg, &process->output_format, sizeof(struct sof_ipc4_audio_format));
 		cfg += sizeof(struct sof_ipc4_audio_format);
+		break;
+	case INIT_PAYLOAD_WITH_BASE_CFG_EXT:
+	{
+		struct sof_ipc4_base_module_cfg_ext *base_cfg_ext;
+		size_t size;
+
+		base_cfg_ext = sof_ipc4_process_add_base_cfg_extn(swidget, &size);
+		if (!base_cfg_ext) {
+			dev_err(sdev->dev, "Error generating up base config extension for %s\n",
+				swidget->widget->name);
```

Comment:
No need for this print as you are printing it already once when looking for the formats